components/uniset/src/utils.rs

```diff
@@ -0,0 +1,64 @@
+use std::{
+    char::MAX,
+    ops::{Bound::*, RangeBounds},
+};
+
+/// Returns whether the vector is sorted ascending non inclusive
+pub fn is_sorted(v: &[u32]) -> bool {
+    v.chunks(2).all(|chunk| chunk[0] < chunk[1])
+}
+
+/// Returns start (inclusive) and end (exclusive) bounds of RangeBounds
+pub fn deconstruct_range(range: impl RangeBounds<char>) -> (u32, u32) {
+    let from = match range.start_bound() {
+        Included(b) => (*b as u32),
+        Excluded(b) => (*b as u32),
```

Comment:
`+ 1` ?

Reply:
That was my original impression but in the `start_bound()` the first value is always inclusive when using the default `RangeBounds` implementations (`Range`, `RangeFrom`, etc).